tree/tree/inc/TTree.h

```diff
@@ -439,7 +445,7 @@ class TTree : public TNamed, public TAttLine, public TAttFill, public TAttMarker
    virtual Long64_t        GetSelectedRows() { return GetPlayer()->GetSelectedRows(); }
    virtual Int_t           GetTimerInterval() const { return fTimerInterval; }
            TBuffer*        GetTransientBuffer(Int_t size);
-   virtual Long64_t        GetTotBytes() const { return fTotBytes; }
+   virtual Long64_t        GetTotBytes() const { std::lock_guard<std::mutex> sentry(fCounterMutex); return fTotBytes; }
```

Comment:
I believe this is correct ( @Dr15Jones - can you confirm); however, I prefer to remove this lock if possible.

What's the best advice for using `std::atomic`?  Maybe use a RAII type like with `GetEntry`?  Would this work:

- When the parallel flush basket starts, set a (transient) flag with a scoped RAII-style object.
- If the transient flag is set, use transient `std::atomic`-based counters.  (This should have no overhead when IMT is disabled).
- When parallel flush ends (no parallel work is going on), add the `std::atomic`-based counters to the existing, non-atomic `fTotBytes` and `fZipBytes` and zero them out.